### PR TITLE
Gist on Test

### PR DESCRIPTION
**Pull request by ryanmccombe**
Issue: https://github.com/bbc/simorgh/issues/9258
Depends On: https://github.com/bbc/simorgh/pull/9281

**Overall change:**
Add Gist block to articles on test

**Code changes:**

- Create Transformer for group blocks.
- Update gist block to make use of new data structure.
- Update `augmentWithTimestamp` to apply the timestamp after the gist block, if it exists
- Add gist block to the page.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/article/handleGroupBlocks/fixture.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (ryanmccombe) [inline src/app/routes/article/handleGroupBlocks/fixture.json:1]: Delete

File: src/app/routes/article/handleGroupBlocks/index.test.js [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      validateBlockExcluded(group),
+    );
+  });
+
+  it('removes non-overview group blocks even when they have a valid gist block structure', () => {
+    runTest(
+      [
+        groupBlock({
+          intentType: 'not-overview',
+          blocks: [textBlock({ blocks: [unorderedListBlock('list')] })],
```
Comment (ryanmccombe) [inline src/app/routes/article/handleGroupBlocks/index.test.js:83]:
```suggestion
          blocks: [textBlock({ blocks: [unorderedListBlock()] })],
```

File: src/app/routes/article/handleGroupBlocks/index.test.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+          blocks: [textBlock({ blocks: [unorderedListBlock()] })],
+        }),
+      ],
+      validateBlockIncluded(media),
+    );
+  });
+
+  it('removes non-overview group blocks', () => {
+    runTest(
+      [groupBlock({ intentType: 'not-overview', blocks: [] })],
```
Comment (ryanmccombe) [inline src/app/routes/article/handleGroupBlocks/index.test.js:73]:
```suggestion
      [groupBlock({ intentType: 'not-overview' })],
```

File: src/app/lib/logger.const.js [hunk 4]
```diff
@@ -58,6 +58,7 @@ const logCodes = {
 
   // Block Types
   UNSUPPORTED_BLOCK_TYPE: 'unsupported_block_type',
+  GIST_TRANSFORMATION_FAILED: 'gist_transformation_failed',
```
Comment (pvaliani) [inline src/app/lib/logger.const.js:61]: Plain and simple but can you explain what this logger does?

Comment (rebeccamcginn) [reply src/app/lib/logger.const.js:61]: These are the constants we use to query our logs. If you have a look at the logging service, you'll see we have queries that make use of these event names to track requests, failures, incomplete data, etc.

This event name in particular is used here: 
https://github.com/bbc/simorgh/pull/9286/files#diff-205fed645f37d0989a7d15a741f0d3cfc86cebf38c24be1eace5a14ebedd6121R124

Whenever the data manipulation fails, we will log the metadata ID along with this constant. We can then track failures on the logging service

We use the Winston library for logging - https://github.com/winstonjs/winston

**Review by nicola-d-bbc — APPROVED** [review]
Lots of good tests!

**Review by pvaliani — APPROVED** [review]
Looks good - quite a lot of code in some individual commits would be my only comment.

Really appreciate the commenting of code as it is helping to follow the logic well.
Thanks for answering my comments!

**Pull request merged**